merlin/dag/executors.py

```diff
@@ -36,8 +39,35 @@ class LocalExecutor:
     An executor for running Merlin operator DAGs locally
     """
 
+    # TODO: Replace `nodes` with `graph` here?
+    def transform_multi(
+        self,
+        data: List[Transformable],
+        schemas,
+        nodes,
+        output_dtypes=None,
+        additional_columns=None,
+        capture_dtypes=False,
+    ):
+        """
+        Transforms multiple dataframes by applying the operators from a collection of Nodes
+        """
+        graphs = [Graph(deepcopy(nodes)[0]).construct_schema(schema) for schema in schemas]
+
+        results = []
+        for graph, data_ in zip(graphs, data):
+            results.append(self.transform(data_, [graph.output_node]))
+
+        return results
+
+    # TODO: Replace `nodes` with `graph` here?
     def transform(
```

Comment:
This method has been long and hard to read for as long as it has existed (well before `Executor`s were a thing yet.) Here it's just broken down into some smaller methods that help explain what's happening via the method names to make it easier to read and think about.